cdap-docs/reference-manual/source/http-restful-api/metrics.rst

```diff
@@ -648,43 +731,139 @@ For aggregates, you cannot specify a time range. As an example, to return the to
+  in the Flow *CountRandom* of the example application *CountRandom*, over the last 5 seconds, you can issue an HTTP
+  POST method::
+
+    POST '<base-url>/metrics/query?tag=namespace:default&tag=app:CountRandom&tag=flow:CountRandom
+      &tag=flowlet:splitter&metric=system.process.events.processed&start=now-5s&count=5'
+
+  This returns a JSON response that has one entry for every second in the requested time interval. It will have
+  values only for the times where the metric was actually emitted (shown here "pretty-printed")::
+
+    {
+      "startTime": 1427225350,
+      "endTime": 1427225354,
+      "series": [
+        {
+          "metricName": "system.process.events.processed",
+          "grouping": { },
+          "data": [
+            {
+              "time": 1427225350,
+              "value": 760
+            },
+            {
+              "time": 1427225351,
+              "value": 774
+            },
+            {
+              "time": 1427225352,
+              "value": 792
+            },
+            {
+              "time": 1427225353,
+              "value": 756
+            },
+            {
+              "time": 1427225354,
+              "value": 766
+            }
+          ]
+        }
+      ]
+    }
+    
+- If a run-ID is not specified, CDAP aggregates the events processed for all the runs of a flow.
+  The resulting timeseries will represent aggregated values for the context specified.
+  Currently, summation is used as the aggregation function.
+  
+  For example, if you query for the ``system.process.events.processed`` metric for a Flow
+  |---| and thus across all Flowlets |---| since this metric was actually emitted at the
+  Flowlet level, the resulting values retrieved will be a sum across all Flowlets of the Flow.
+
+- If you want the number of input objects processed across all Flowlets of a Flow, you 
+  address the Metrics API at the Flow context::
+
+    POST '<base-url>/metrics/query?tag=namespace:default.app:CountRandom?tag=flow:CountRandom?tag=flowlet:*
+      &metric=system.process.events.processed&start=now-5s&count=5'
+
+  Similarly, you can address the context of all Flows of an Application, an entire Application, 
+  or the entire  namespace of a CDAP instance::
+
+    POST '<base-url>/metrics/query?tag=namespace:default&tag=app:CountRandom&tag=flow:*
+      &metric=system.process.events.processed&start=now-5s&count=5'
+
+    POST '<base-url>/metrics/query?tag=namespace:default&tag=app:CountRandom
+      &metric=system.process.events.processed&start=now-5s&count=5'
+
+    POST '<base-url>/metrics/query?tag=namespace:default
+      &metric=system.process.events.processed&start=now-5s&count=5'
+
+- To request user-defined metrics instead of system metrics, specify ``user`` instead of 
+  ``cdap`` in the URL and specify the user-defined metric at the end of the request.
+
+  For example, to request a user-defined metric for the *HelloWorld* Application's *WhoFlow* Flow::
```

Comment:
to query the user-defined metric ``names.bytes`` in the *HelloWorld* Application